spring-boot-project/spring-boot/src/test/java/org/springframework/boot/jdbc/metadata/HikariDataSourcePoolMetadataTests.java

```diff
@@ -57,4 +62,12 @@ private HikariDataSource createDataSource(int minSize, int maxSize) {
 		return dataSource;
 	}
 
+	@Test
+	void getIdle() {
+		// Make sure the pool is initialized
+		JdbcTemplate jdbcTemplate = new JdbcTemplate(getDataSourceMetadata().getDataSource());
+		jdbcTemplate.execute((ConnectionCallback<Void>) (connection) -> null);
+		assertThat(getDataSourceMetadata().getIdle()).isEqualTo(Integer.valueOf(1));
```

Comment:
This assumes that an Hikari connection pool with min=0; max=2 has 1 initial connection ?
The other connection pools seem to default to 0 - or is that because the other connection pools are _not initiliazed_ ?

Reply:
yes, it's because the other connection pools are not initialized

Reply:
I guess I need to add the same tests with initialization pool for other connection pools, what do you think about it?

Reply:
We should configure the pool explicitly so that the idle connection stays in the pool indeed.

Reply:
sorry, I'm not sure that I catch you, Is this way of initializing pool good for the tests?

Reply:
We could use a `HikariDataSource dataSource = createDataSource(0, 2);` in the test itself. If there is a setting that we could set to increase the time an idle connection stays in the pool, that would be nice too as we ensure that our test is asserting the conditions as we expect them.

Reply:
ok, I have got it.